extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -231,40 +193,28 @@ class CornerstoneViewportService extends PubSubService
   /**
    * Uses the renderingEngine to enable the element for the given viewport index
    * and sets the displaySet data to the viewport
-   * @param {*} viewportIndex
+   * @param {*} viewportId
    * @param {*} displaySet
    * @param {*} dataSource
    * @returns
    */
   public setViewportData(
-    viewportIndex: number,
+    viewportId: string,
     viewportData: StackViewportData | VolumeViewportData,
     publicViewportOptions: PublicViewportOptions,
     publicDisplaySetOptions: DisplaySetOptions[],
     presentations?: Presentations
   ): void {
     const renderingEngine = this.getRenderingEngine();
-    const viewportId =
-      publicViewportOptions.viewportId || this.getViewportId(viewportIndex);
-    if (!viewportId) {
-      throw new Error('Must define viewportId externally');
-    }
 
-    const viewportInfo = this.viewportsById.get(viewportId);
+    const viewportInfo = this.viewportsById.get(
+      publicViewportOptions.viewportId
```

Comment:
I expected that `viewportId` would be used here. Why `publicViewportOptions.viewportId` instead?